### DISCUSSION TITLE
Cannot clone Airflow repo

### DISCUSSION THREAD

**Discussion by hansmbakker** [apache/airflow#25285] [General]
### Apache Airflow version

2.3.3 (latest released)

### What happened

The `git clone` command failed because of files in [/tests/www/test_logs/dag_for_testing_log_view](https://github.com/apache/airflow/tree/main/tests/www/test_logs/dag_for_testing_log_view) having invalid names:
```
PS C:\client-repos\apache> git clone https://github.com/apache/airflow
Cloning into 'airflow'...
remote: Enumerating objects: 249170, done.
remote: Counting objects: 100% (166/166), done.
remote: Compressing objects: 100% (122/122), done.
remote: Total 249170 (delta 65), reused 93 (delta 43), pack-reused 249004
Receiving objects: 100% (249170/249170), 173.14 MiB | 8.05 MiB/s, done.
Resolving deltas: 100% (188121/188121), done.
error: invalid path 'tests/www/test_logs/dag_for_testing_log_view/scheduled__2017-09-01T00:00:00+00:00/task_for_testing_log_view/1.log'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

### What you think should happen instead

Either it should checkout 

### How to reproduce

Use Windows 11 with the Terminal app in Powershell mode.

Run `git clone https://github.com/apache/airflow`

### Operating System

Windows

### Deployment

Other

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by hansmbakker** [comment]
Workaround: cloning inside WSL works, but I believe that using filenames containing characters that are unsafe in a path is not a best practice.

**Reply by potiuk** [reply]
> Workaround: cloning inside WSL works, but I believe that using filenames containing characters that are unsafe in a path is not a best practice.

Feel free to make PR changing it (including fixing unit tests depending on it as part of #10388 . Having Windows support is on our roadmap and help from users like you who use Windows and would like to make it works for them is most welcome. This is the best contribution you can make and paying back for the free software you use.

**Reply by potiuk** [reply]
You are - actually - one of the best people on earth to make such change, since you experienced it and want it to change and have an easy way to test it. You are absolutely welcome to make it happen.

**Reply by hansmbakker** [reply]
Sorry, but I am not an Airflow user - I merely wanted to clone the repo to search something inside the repo easier. For now, the workaround I shared is sufficient and by reporting the issue I documented it.
Also, I do not know whether those folders with their specific name format are there for a reason.
Therefore I think it is more efficient to leave this to a person with more background in Airflow.

**Reply by potiuk** [reply]
Up to you :). Maybe someone will do it in the meantime as part of #10388

**Comment by potiuk** [comment]
Don't clone Airflow in Windows. Airlfow does not work on Windows as development environment (yet). There is #10388  in progress and any help is welcome to implement what's needed there.
